src/navigation/checkout/PaymentMethodEdenred.js

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { Button, Text, VStack, HStack, Box, Skeleton } from 'native-base';
+import { useTranslation } from 'react-i18next';
+import { authorize } from 'react-native-app-auth';
+import 'react-native-get-random-values';
+import { v4 as uuidv4 } from 'uuid';
+import { useNavigation } from '@react-navigation/native';
+import Config from 'react-native-config';
+import _ from 'lodash';
+import {
+  updateEdenredCredentials,
+  loadPaymentDetails,
+  checkout,
+} from '../../redux/Checkout/actions';
+import {
+  selectCart,
+} from '../../redux/Checkout/selectors';
+import CreditCard from './components/CreditCard';
+import { formatPrice } from '../../utils/formatting'
+import HeaderHeightAwareKeyboardAvoidingView from '../../components/HeaderHeightAwareKeyboardAvoidingView';
+
+const Edenred = ({ clientId, baseURL, cart, paymentDetailsLoaded, loadPaymentDetails, paymentDetails, errors, checkout, updateEdenredCredentials, authorizationEndpoint }) => {
+
+  const navigation = useNavigation();
+  const { t } = useTranslation();
+
+  useEffect(() => {
```

Comment:
Looks like it's too much for `useEffect`. I'd suggest moving it into `actions.js`